examples/sn-dms-demo/src/components/UsersAndGroups/UserSelector/UserListItem.tsx

```diff
@@ -0,0 +1,136 @@
+import IconButton from '@material-ui/core/IconButton'
+import ListItemIcon from '@material-ui/core/ListItemIcon'
+import ListItemText from '@material-ui/core/ListItemText'
+import MenuItem from '@material-ui/core/MenuItem'
+import withStyles from '@material-ui/core/styles/withStyles'
+import { User } from '@sensenet/default-content-types'
+import { Icon, iconType } from '@sensenet/icons-react'
+import { Actions } from '@sensenet/redux'
+import * as React from 'react'
+import { connect } from 'react-redux'
+import { RouteComponentProps, withRouter } from 'react-router-dom'
+import { rootStateType } from '../../../store/rootReducer'
+import { selectUser } from '../../../store/usersandgroups/actions'
+
+const styles = {
+  listItem: {
+    listStyleType: 'none',
+    borderTop: 'solid 1px #2080aa',
+    padding: '12px 12px 12px 0px',
+  },
+  listItemRoot: {
+    padding: 0,
+  },
+  primary: {
+    fontFamily: 'Raleway ExtraBold',
+    fontSize: 15,
+    lineHeight: '24px',
+    color: '#fff',
+    background: 'none',
+    padding: 0,
+    '&:hover': {
+      backgroundColor: 'transparent',
+    },
+  },
+  secondary: {
+    color: '#fff',
+    fontFamily: 'Raleway SemiBold',
+    fontStyle: 'italic',
+    fontSize: 11,
+  },
+  icon: {
+    margin: 0,
+    color: '#fff',
+  },
+  iconButton: {
+    margin: 0,
+    padding: 0,
+    '&:hover': {
+      backgroundColor: 'transparent',
+    },
+  },
+  followedIconButton: {
+    margin: 0,
+    padding: 0,
+    color: '#ffeb3b',
+  },
+}
+
+interface UserListItemProps extends RouteComponentProps<any> {
+  selected: boolean
+  user: User
+  userName: string
+  closeDropDown: (open: boolean) => void
+}
+
+const mapStateToProps = (state: rootStateType) => {
+  return {
+    userName: state.sensenet.session.user.userName,
+    options: state.sensenet.currentitems.options,
+    users: state.dms.usersAndGroups.user.selected,
+  }
+}
+
+const mapDispatchToProps = {
+  selectUser,
+  loadContent: Actions.loadContent,
+  fetchContent: Actions.requestContent,
+}
+
+interface UserListItemState {
+  selected: boolean
+}
+
+class UserListItem extends React.Component<
+  { classes: any } & ReturnType<typeof mapStateToProps> & typeof mapDispatchToProps & UserListItemProps,
+  UserListItemState
+> {
+  public state = {
+    selected: this.props.selected,
```

Comment:
You should use the props.selected value instead of using it as a state because this way it is not working as expected. It is not going to be updated when the props change.
The simplest solution is to use the props.selected state. The other one is to use getderivedstatefromprops.